### PR TITLE
fix: disallow expressions inside literal lists/maps

### PR DESCRIPTION
**Pull request by sargunv**
it seems literal lists can't have expressions inside, so updating the expressions DSL typing

### CODE REVIEW

**Comment by sargunv** [comment]
> [!WARNING]
> <b>This pull request is not mergeable via GitHub because a downstack PR is open. Once all requirements are satisfied, merge this PR as a stack <a href="https://app.graphite.dev/github/pr/sargunv/maplibre-compose/141?utm_source=stack-comment-downstack-mergeability-warning" >on Graphite</a>.</b>
> <a href="https://graphite.dev/docs/merge-pull-requests">Learn more</a>

* **#141** <a href="https://app.graphite.dev/github/pr/sargunv/maplibre-compose/141?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈 <a href="https://app.graphite.dev/github/pr/sargunv/maplibre-compose/141?utm_source=stack-comment-view-in-graphite" target="_blank">(View in Graphite)</a>
* **#140** <a href="https://app.graphite.dev/github/pr/sargunv/maplibre-compose/140?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* `main`


<h2></h2>

This stack of pull requests is managed by <a href="https://graphite.dev?utm-source=stack-comment"><b>Graphite</b></a>. Learn more about <a href="https://stacking.dev/?utm_source=stack-comment">stacking</a>.

File: lib/maplibre-compose/src/commonMain/kotlin/dev/sargunv/maplibrecompose/core/expression/ExpressionsDsl.kt [hunk 1]
```diff
@@ -69,6 +69,11 @@ public object ExpressionsDsl {
   public fun const(padding: PaddingValues.Absolute): Expression<PaddingValue> =
     Expression.ofPadding(padding)
 
+  /** Creates a literal expression for a list of strings. */
+  @JvmName("constStringList")
+  public fun const(list: List<String>): Expression<ListValue<StringValue>> =
+    Expression.ofList(listOf("literal", list)).cast()
```
Comment (sargunv) [inline lib/maplibre-compose/src/commonMain/kotlin/dev/sargunv/maplibrecompose/core/expression/ExpressionsDsl.kt:75]: do we need one for numbers? we already have offset, dpoffset, and padding. Unsure if there's other lists of numbers around

**Comment by sargunv** [comment]
### Merge activity

* **Dec 18, 3:34 PM PST**: A user merged this pull request with [Graphite](https://app.graphite.dev/github/pr/sargunv/maplibre-compose/141).

**Comment by westnordost** [comment]
They can't? That makes things easier, certainly. Only on MLN or also on ML web?

**Comment by sargunv** [comment]
Unsure about web, but yeah on iOS at least I was getting errors on things like `"text-offset": ["literal", [["var", "x"], ["var", "y"]]]` because it was expecting an array of 2 numbers but instead got an array of size 3.

Updating it to just `[["var", "x"], ["var", "y"]]` errors out because it wants an array of two numbers but instead got an array of two string arrays.

**Comment by westnordost** [comment]
So, this affects DpOffset (etc), too? I.e. anything that is nested in any way can only be an expression on the top level?

**Comment by sargunv** [comment]
Our `DpOffset` const function already disallowed expressions inside the list (because we accepted a DpOffset object itself rather than a tuple of Dp expressions)

Unsure if it affects a string list (used for font names), gotta confirm that (this PR assumes it doesn't).

Also unsure if it affects `textVariableAnchorOffset`, working on typing that next so I'll find out

**Pull request merged**